### PR TITLE
Re-factor code corresponding to #41 issue in Nailgun

### PR DESCRIPTION
**Pull request by oshtaier**
Current fix should be merged right after SatelliteQE/nailgun#165 get to the master branch
Also, all tests need to be executed. List of tests that are run at that moment:
```
nosetests tests/foreman/api/test_contentview.py
...........................................SSSSSSSSSSSSSSSSSSSSSSS...
----------------------------------------------------------------------
Ran 69 tests in 2192.104s

OK (SKIP=23)
```
```
nosetests tests/foreman/smoke/test_api_smoke.py -m test_smoke
.
----------------------------------------------------------------------
Ran 1 test in 175.020s

OK
```
```
nosetests tests/foreman/api/test_contentviewversion.py -m test_negative_promote_
..
----------------------------------------------------------------------
Ran 2 tests in 5.735s

OK
```
```
nosetests tests/foreman/api/test_contentviewversion.py -m test_delete_version_non_default
.
----------------------------------------------------------------------
Ran 1 test in 63.522s

OK
```
```
nosetests tests/foreman/smoke/test_api_smoke.py -m test_end_to_end
.
----------------------------------------------------------------------
Ran 1 test in 212.078s

OK
```
```
nosetests tests/foreman/ui/test_contentviews.py -m test_delete_version_non_default
.
----------------------------------------------------------------------
Ran 1 test in 140.471s

OK
```
```
nosetests tests/foreman/ui/test_contentviews.py -m test_delete_version_with_ak
.
----------------------------------------------------------------------
Ran 1 test in 130.853s

OK
```
```
nosetests tests/foreman/api/test_contentviewfilter.py -m test_create_rpm_cvf_with_different_names
.......
----------------------------------------------------------------------
Ran 7 tests in 85.158s

OK
```
```
nosetests tests/foreman/api/test_contentviewfilter.py -m test_create_cvf_with_multiple_repos_with_docker
.
----------------------------------------------------------------------
Ran 1 test in 48.061s

OK
```
```
nosetests tests/foreman/api/test_contentviewfilter.py -m test_update_cvf_with_new_repo
..
----------------------------------------------------------------------
Ran 2 tests in 100.559s

OK
```
```
nosetests tests/foreman/api/test_contentviewfilter.py -m test_update_cvf_with_multiple_repos
.
----------------------------------------------------------------------
Ran 1 test in 136.849s

OK
```
```
nosetests tests/foreman/api/test_contentviewfilter.py -m test_update_cvf_with_repo_with_docker
.
----------------------------------------------------------------------
Ran 1 test in 50.284s

OK
```
```
nosetests tests/foreman/ui/test_contentsearch.py -m test_search_content_view
.
----------------------------------------------------------------------
Ran 1 test in 97.228s

OK
```
```
nosetests tests/foreman/api/test_product.py -m test_repositoryset_enable
.
----------------------------------------------------------------------
Ran 1 test in 87.175s

OK
```
```
nosetests tests/foreman/api/test_product.py -m test_repositoryset_disable
.
----------------------------------------------------------------------
Ran 1 test in 108.823s

OK
```
```
nosetests tests/foreman/api/test_subscription.py
...
----------------------------------------------------------------------
Ran 3 tests in 213.186s

OK
```
```
nosetests tests/foreman/ui/test_syncplan.py -m test_positive_update_2
...............
----------------------------------------------------------------------
Ran 15 tests in 515.254s

OK
```
```
nosetests tests/foreman/ui/test_syncplan.py -m test_positive_delete_1
.......
----------------------------------------------------------------------
Ran 7 tests in 293.525s

OK
```

### CODE REVIEW

File: tests/foreman/ui/test_activationkey.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def cv_publish_promote(self, name, env_name, repo_id, org_id=None):
         env_attrs = entities.LifecycleEnvironment(
             name=env_name,
             organization=org_id or self.org_id
-        ).create_json()
+        ).create()
```
Comment (omaciel) [inline tests/foreman/ui/test_activationkey.py]: :+1:

File: tests/foreman/ui/test_contentviews.py [hunk 2]
```diff
@@ -1255,14 +1253,14 @@ def test_delete_version_non_default(self):
...
-        cv.set_repository_ids([repo.id])
+        cv.set_repository_ids({u'repository_ids': [repo.id]})
         cv.publish()
 
         cv_info = cv.read_json()['versions'][0]
         version = 'Version {0}'.format(cv_info['version'])
         cvv = entities.ContentViewVersion(id=cv_info['id'])
         lc_env = entities.LifecycleEnvironment(organization=org).create()
-        cvv.promote(lc_env.id)
+        cvv.promote({'environment_id': lc_env.id})
```
Comment (omaciel) [inline tests/foreman/ui/test_contentviews.py]: Not a blocker but you may want to always make your strings `unicode` like some of your other changes.

**Comment by omaciel** [comment]
ACK (please be aware that this PR cannot yet be merged)

File: tests/foreman/ui/test_contentviews.py (hunk 2 above)
Comment (oshtaier) [inline tests/foreman/ui/test_contentviews.py]: Sure, thanks!

**Comment by oshtaier** [comment]
Added, probably :), all test results, but anyway, will track failures on Jenkins.

**Comment by sthirugn** [comment]
@oshtaier rebase please?

**Comment by oshtaier** [comment]
@sthirugn Done. Thanks!

**Comment by sthirugn** [comment]
@oshtaier @elyezer @omaciel @abalakh 
In this PR, there are 
* 24 instances of u'environment_id'
* 10 instances of u'name'
* 8 instances of u'author'
and so on.
20+ instance usage of the same constant is not an efficient progamming method IMO.  My comment is not a blocker for this PR but we surely need to work to optimize this.

**Comment by oshtaier** [comment]
Merge only in case nailgun PRs merged into master!

**Comment by Ichimonji10** [comment]
See SatelliteQE/nailgun#171.

**Comment by oshtaier** [comment]
Working on changes to current PR

**Comment by elyezer** [comment]
Update subscriptions listing when available:

- [ ] tests/foreman/rhai/test_rhai.py line 38 listing subs
- [ ] tests/foreman/smoke/test_api_smoke.py line 1026 listing subs
- [ ] tests/foreman/ui/test_activationkey.py line 1343 listing subs

**Comment by oshtaier** [comment]
Fixed most of the issues

File: tests/foreman/rhai/test_rhai.py [hunk 3]
```diff
@@ -35,15 +37,15 @@ def setUpClass(cls):  # noqa
         # Walk through the list of subscriptions.
         # Find the "Red Hat Employee Subscription" and attach it to the
         # recently-created activation key.
-        for subscription in org.subscriptions():
-            if subscription['product_name'] == DEFAULT_SUBSCRIPTION_NAME:
+        for subs in sub.search():
+            if subs.read_json()['product_name'] == DEFAULT_SUBSCRIPTION_NAME:
```
Comment (Ichimonji10) [inline tests/foreman/rhai/test_rhai.py]: FYI, you can do something like this:

```python
for result in sub.search_json()['results']:
    if result['product_name'] == DEFAULT_SUBSCRIPTION_NAME:
```

But don't consider this comment a blocker. I'm just letting you know of an alternate way of doing things that doesn't require calling `read_json()` on each result.

**Comment by oshtaier** [comment]
All necessary changes are present in Nailgun

File: tests/foreman/rhai/test_rhai.py (hunk 3 above)
Comment (oshtaier) [inline tests/foreman/rhai/test_rhai.py]: Thank you as I really wanted to find alternative! :+1:

**Comment by Ichimonji10** [comment]
Looks great. ACK!

**Comment by elyezer** [comment]
ACK, will leave for @sthirugn take a look before merging.

File: tests/foreman/api/test_contentview.py [hunk 4]
```diff
@@ -183,7 +182,8 @@ def test_cv_associate_composite_dupe_repos_negative(self):
         content_view = entities.ContentView(organization=org).create()
         self.assertEqual(len(content_view.repository), 0)
         with self.assertRaises(HTTPError):
-            content_view.set_repository_ids([yum_repo.id, yum_repo.id])
+            content_view.repository = [yum_repo, yum_repo]
+            content_view.update(['repository'])
```
Comment (sthirugn) [inline tests/foreman/api/test_contentview.py]: Nice!

File: tests/foreman/smoke/test_api_smoke.py [hunk 5]
```diff
@@ -879,22 +879,24 @@ def test_smoke(self):
         ).create()
 
         # step 2.8: Associate YUM repository to new content view
-        content_view.set_repository_ids([repo1.id])
+        content_view.repository = [repo1]
```
Comment (sthirugn) [inline tests/foreman/smoke/test_api_smoke.py]: so in general, does this append the repo to content view or replace existing repos with the supplied one?

**Comment by sthirugn** [comment]
ACK

File: tests/foreman/smoke/test_api_smoke.py (hunk 5 above)
Comment (Ichimonji10) [inline tests/foreman/smoke/test_api_smoke.py]: I believe we're replacing whatever repositories the content view already has. You can append with something like this:

```python
content_view = ContentView(...).read()
content_view.repository.append(second_repo)
content_view = content_view.update(['repository'])
```

**Pull request merged**